bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/api/IntesisBoxSocketApi.java

```diff
@@ -0,0 +1,232 @@
+        this.ipAddress = ipAddress;
+        this.port = port;
+    }
+
+    private class IntesisSocket {
+        final Socket socket;
+
+        public IntesisSocket() throws UnknownHostException, IOException {
+            socket = new Socket();
+            SocketAddress tcpSocketAddress = new InetSocketAddress(ipAddress, port);
+            socket.connect(tcpSocketAddress);
+        }
+
+        public void close() throws IOException {
+            socket.close();
+        }
+    }
+
+    @SuppressWarnings("null")
+    public void openConnection() throws IOException {
+        closeConnection();
+
+        logger.debug("openConnection(): Connecting to IntesisBox ");
+
+        IntesisBoxChangeListener listener = this.changeListener;
+        if (listener != null) {
+            logger.debug("IntesisBox listener added");
+        }
+
+        tcpSocket = new IntesisSocket();
+        tcpOutput = new OutputStreamWriter(tcpSocket.socket.getOutputStream(), "US-ASCII");
+        tcpInput = new BufferedReader(new InputStreamReader(tcpSocket.socket.getInputStream()));
+
+        Thread tcpListener = new Thread(new TCPListener());
+        tcpListener.start();
+
+        setConnected(true);
+        if (listener != null) {
+            listener.connectionStatusChanged(ThingStatus.ONLINE);
+        }
+    }
+
+    @SuppressWarnings("null")
+    public void closeConnection() {
+        try {
+            if (tcpSocket != null) {
+                logger.debug("closeConnection(): Closing Socket!");
+                tcpSocket.close();
+                tcpSocket = null;
+            }
+            if (tcpInput != null) {
+                logger.debug("closeConnection(): Closing Output Writer!");
+                tcpInput.close();
+                tcpInput = null;
+            }
+            if (tcpOutput != null) {
+                logger.debug("closeConnection(): Closing Input Reader!");
+                tcpOutput.close();
+                tcpOutput = null;
+            }
+
+            setConnected(false);
+            logger.debug("closeConnection(): Closed TCP Connection!");
+        } catch (IOException ioException) {
+            logger.debug("closeConnection(): Unable to close connection - {}", ioException.getMessage());
+        } catch (Exception exception) {
+            logger.debug("closeConnection(): Error closing connection - {}", exception.getMessage());
+        }
+    }
+
+    private class TCPListener implements Runnable {
+        private final Logger logger = LoggerFactory.getLogger(TCPListener.class);
+
+        /**
+         * Run method. Runs the MessageListener thread
+         */
+        @Override
+        public void run() {
+            try {
+                while (isConnected()) {
+                    String message = read();
+                    try {
+                        logger.trace("Calling readMessage()");
+                        readMessage(message);
+                    } catch (Exception e) {
```

Comment:
What type of exception do you expect here? Can you specify the concrete type? Same for below.

Reply:
removed try catch here

Reply:
removed try catch here